internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -204,23 +203,26 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 			//       Though replay would work just fine, and result in a no-op,
 			//       it's nice to not have to run through its entirely. Temporal
 			//       just seems to ignore it...
-			z.Info("already completed")
+			l.Info("already completed")
 			return nil, nil
 		}
 
 		sessionsCreatedCounter.Add(ctx, 1)
 
-		z.Info("session workflow: started")
-		prints, err = runWorkflow(wctx, z, ws, data, params.Debug)
+		l.Info("session workflow: started", zap.Int32("attempt", wi.Attempt))
+		startTime := time.Now()
+		prints, err = runWorkflow(wctx, l, ws, data, params.Debug)
+		duration = time.Since(startTime)
+		sessionDurationHistogram.Record(ctx, duration.Milliseconds())
```

Comment:
Does this take into account replay? 
Failure of code and restart of the service?

Reply:
could u plz give me exact use case that you think we are not handling?

Reply:
efi means that when the workflow is being replayed, this will count the execution twice. you _could_ ignore replays, just wrap with a `workflow.IsReplay(wctx)`, but I'm not sure that's the right approach.

we'll need to discuss this in sync.

Reply:
when workflow is replayed it will be different session, no?

Reply:
rewritten

Reply:
maybe we need to add a dimention on the final state
otherwise canclled/errored sessions might give us outliers on actual completed runs

Reply:
we take the latest time of a final state and compare to start time
this is basically full session from AK's perspective, not temporal, which is better for us
resolving